### PR TITLE
Update for parity, and do some makefile/buildkite refinement to be clearer about publishing vs testing

### PR DESCRIPTION
**Pull request by sinneduy**
Parity commit: https://github.com/awslabs/amazon-eks-ami/compare/7f6c8cb3597e17f6e5f7df96d12bccf5604dc909...fe39deb2811672fb3279f3ecded3c88df73979f6

Unfortunately, I missed a [bunch of commits from January 30th](https://github.com/awslabs/amazon-eks-ami/compare/7f6c8cb3597e17f6e5f7df96d12bccf5604dc909...dedf0960383f75974596d4512309a3ac5aeebc96) initially, so the last commit adds those in.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -1,20 +1,78 @@
 steps:
-  - label: ':ubuntu: Build ubuntu AMI'
-    command: 'docker run -it -v $(pwd)/install-worker.sh:/install-worker.sh -v $(pwd)/files:/files -v $(pwd)/eks-worker-ubuntu.json:/eks-worker-ubuntu.json -e VPC -e SUBNET hashicorp/packer build -var "vpc_id=$VPC" -var "subnet_id=$SUBNET" eks-worker-ubuntu.json'
+  - label: 'validate packer template'
+    command: 'docker build . -t gusto/packer && docker run -v $(pwd):/packer -e VPC_ID -e SUBNET_ID gusto/packer make validate'
     agents:
       queue: 'eks-ubuntu-packer'
+  - label: ':ubuntu: Test EKS ubuntu AMI for 1.10'
+    command: 'docker build . -t gusto/packer && docker run -v $(pwd):/packer -e VPC_ID -e SUBNET_ID gusto/packer make 1.10'
```
Comment (juleea) [inline .buildkite/pipeline.yml]: should this be `gusto:packer`?
image

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -1,20 +1,78 @@
...
+    agents:
+      queue: 'eks-ubuntu-packer'
+    branches: '!master'
+    retry:
+      automatic:
```
Comment (juleea) [inline .buildkite/pipeline.yml]: does this have a history of failing nondeterministically?

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -1,20 +1,78 @@
...
+  - label: ':ubuntu: Test EKS ubuntu AMI 1.12'
+    command: 'docker build . -t gusto/packer && docker run -v $(pwd):/packer -e VPC_ID -e SUBNET_ID gusto/packer make 1.12'
+    agents:
+      queue: 'eks-ubuntu-packer'
+    branches: '!master'
+    retry:
+      automatic:
+        - limit: 3
+          exit_status: "*"
+  - label: ':ubuntu: Test EKS ubuntu AMI 1.12'
```
Comment (juleea) [inline .buildkite/pipeline.yml]: should this be AMI latest?

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (juleea) [reply .buildkite/pipeline.yml]: oh looks like yes: https://buildkite.com/gusto/packer-ubuntu-eks-ami/builds/76

**Comment by sinneduy** [comment]
@juleea I took some liberties in the following places:
https://github.com/Gusto/ubuntu-eks-ami/pull/4/files#diff-b67911656ef5d18c4ae36cb6741b7965R60
https://github.com/Gusto/ubuntu-eks-ami/pull/4/files#diff-4de22a3515c1c580a6b6a45c86b7289cL121
https://github.com/Gusto/ubuntu-eks-ami/pull/4/files#diff-4de22a3515c1c580a6b6a45c86b7289cR183

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (sinneduy) [reply .buildkite/pipeline.yml]: I don't believe so.  the `name` this is referring to also includes the organization, so something like `gusto/test` or [`xbblabs/parsedmarc`](https://hub.docker.com/r/xbblabs/parsedmarc) would be the `name`.  We only see non-slash names when an image is the official docker image

The `tag` instead refers to the "version" like you could see on the [ruby documentation](https://hub.docker.com/_/ruby)

That being said, I don't actually intend to upload that image.  We could, but I think I'd prefer not to for now, I just want to build it so I can run the make command with a packer binary

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (sinneduy) [reply .buildkite/pipeline.yml]: unfortunately =/.  The piece that always fails is installing a number of apt libraries that are installed in the first step

File: .buildkite/pipeline.yml (hunk 3 above)
Comment (sinneduy) [reply .buildkite/pipeline.yml]: yep!

File: install-worker.sh [hunk 4]
```diff
@@ -118,45 +155,83 @@ sudo mkdir -p /var/lib/kubernetes
 sudo mkdir -p /var/lib/kubelet
 sudo mkdir -p /opt/cni/bin
 
-CNI_VERSION=${CNI_VERSION:-"v0.6.0"}
```
Comment (juleea) [inline install-worker.sh]: do you know why this was here? was it something you added?

Comment (sinneduy) [reply install-worker.sh]: I believe this was for parity:
https://github.com/awslabs/amazon-eks-ami/commit/ca5559a72ce4700ba26ef1e5890404a35ec650b3#diff-4de22a3515c1c580a6b6a45c86b7289cL96

I think it makes sense though - in this commit and in my commit you can see that at the top of the `install-worker.sh` we now ensure that the `CNI_VERSION` variable is set

Comment (juleea) [reply install-worker.sh]: ah gotcha - your changes are to the download resource paths, not this specific line

Comment (sinneduy) [reply install-worker.sh]: Yeah, I think it is a "grouped" change. to align with their changes

File: install-worker.sh [hunk 5]
```diff
@@ -118,45 +155,83 @@ sudo mkdir -p /var/lib/kubernetes
...
 EOF"
-sudo apt-get update
-sudo apt-get install -y kubelet=$KUBE_VERSION-00 kubectl=$KUBE_VERSION-00
+
+sudo apt-get update -y
+
+# This hack is necessary for:
+# https://github.com/kubernetes/kubernetes/issues/69489
+# EKS in Amazon Linux 2 bypasses this because it installs kubelet by downloading a binary
+if [ ${KUBERNETES_VERSION:0:4} == "1.10" ]; then
```
Comment (juleea) [inline install-worker.sh]: should this be `<`? a comment on GH issue suggests this is an issue for 1.9.*

**Comment by juleea** [comment]
@sinneduy thanks, that helps. I added a few questions but overall looks reasonable to me.

File: install-worker.sh (hunk 5 above)
Comment (sinneduy) [reply install-worker.sh]: EKS doesn't support `1.9`, so I figured we wouldn't need builds.  I was actually considering just not doing builds for `1.10` also, but I figured I'd leave it around as long as you could bring up an EKS master for `1.10`

Comment (sinneduy) [reply install-worker.sh]: Also looks like `1.10` is being deprecated in late July:
https://github.com/aws/containers-roadmap/issues/300

**Pull request merged**